install/linux/usr/bin/odemis-select-mic-start

```diff
@@ -56,26 +119,66 @@ mkdir $HOME/.local/share/odemis/log 2>/dev/null
 parse_logfile=$HOME/.local/share/odemis/log/odemis-select-mic-start-parse.log
 > $parse_logfile
 
-for fn in "${args[@]}"; do
-    # add a new header to the logfile
-    printf "$(date) Parsing file: $fn\n" >> $parse_logfile
-    # extract the microscope name by using this small Python script
-    mic_name="$(python3 -c "import sys; import yaml; from odemis.odemisd import modelgen; \
-        f = open('$fn'); \
-        d = yaml.load(f, modelgen.SafeLoader); \
-        print(next(n for n, d in d.items() if d.get('class') == 'Microscope'))" 2>>$parse_logfile)"
-    mic_list+=("$fn")
+process_file() {
+    local fn="$1"
+
+    # Add a new header to the logfile
+    printf "$(date) Parsing file: $fn\n" >> "$parse_logfile"
+
+    # Extract the microscope name and role using a small Python script
+mic_name=$(python3 <<EOF 2>>"$parse_logfile"
+import sys
+import yaml
+from odemis.odemisd import modelgen
+
+f = open('$fn')
+d = yaml.load(f, modelgen.SafeLoader)
+mic_name = next(n for n, d in d.items() if d.get('class') == 'Microscope')
+print(mic_name)
+EOF
+    )
+
+    mic_list+=("$fn")  # The dialog has a hidden column, which is the filepath
+
     if test "$mic_name" == ""; then
         # save the file name in the error list
         error_filenames+=("$(basename ${fn})")
         mic_name=">>ERROR parsing $(basename ${fn})"
         # add a newline to the logfile
         echo "" >> $parse_logfile
     fi
+
     mic_list+=("$mic_name")
+}
+
+total_files=${#target_files[@]}
+current_file=0
+
+# Processing a large selection of simulators can take quite a while,
+# so giving visual feedback of the progress gives the user clarity
+# that the script is still running. Using file descriptor so we are
+# still able to append to mic_list.
+exec 3> >(zenity --progress \
+           --title="Validation" \
+           --text="Validating Simulators..." \
```

Comment:
They are not always simulators. These are just microscope files. Also, it's "validating" sounds like it's checking that the user didn't do a mistake. Typically, the files are fine, and the main reason we are reading them is to find out their name. So I'd say "Analysing microscope configurations..."